src/pages/signUp/SignUpPage.tsx

```diff
@@ -77,6 +94,11 @@ export default function SignUpPage() {
               "md:pr-24"
             )}
           >
+            {emailStatus.error && (
```

Comment:
Probably considering that many of your TextInput shows errors I would have created a component and reuse it like for instance: 

```javascript
export const TextInputWithError = (label, validatorsFunc, id, ...textInputProps) => {
  const [errorStatus, validation] = useValidation([...validatorsFunc]);

  return (
    <>
      {errorStatus.error && (
        <ValidationError className={cx("col-span-3")}>
          {errorStatus.error}
        </ValidationError>
      )}
      <label className={cx("col-span-2", "sm:col-span-1")} htmlFor={id}>
        {label}
      </label>
      <TextInput id={id} validation={validation} {...textInputProps} />
    </>
  );
};

```

Reply:
Thanks for the suggestion, but I wouldn't do that, because I don't want to mix up layout and logic. The `TextInput` and `ValidationError` components are pure layout components. Note how I didn't put the `col-span` classes in those – these are specific to the login / sign up forms. If I refactored the error and input into a `TextInputWithError` component, I'd have to figure our a way to keep the `col-span` classes out of it.

Code duplication in this case is a good thing, in my opinion.